homeassistant/components/steam_online/coordinator.py

```diff
@@ -29,42 +31,40 @@ def __init__(self, hass: HomeAssistant) -> None:
             update_interval=timedelta(seconds=30),
         )
         self.game_icons: dict[int, str] = {}
-        self.player_interface: INTMethod = None
-        self.user_interface: INTMethod = None
-        steam.api.key.set(self.config_entry.data[CONF_API_KEY])
+        key.set(self.config_entry.data[CONF_API_KEY])
 
     def _update(self) -> dict[str, dict[str, str | int]]:
         """Fetch data from API endpoint."""
-        accounts = self.config_entry.options[CONF_ACCOUNTS]
-        _ids = list(accounts)
-        if not self.user_interface or not self.player_interface:
-            self.user_interface = steam.api.interface("ISteamUser")
-            self.player_interface = steam.api.interface("IPlayerService")
+        _ids = list(self.config_entry.options[CONF_ACCOUNTS])
+        reg = er.async_get(self.hass)
         if not self.game_icons:
+            _interface = interface("IPlayerService")
             for _id in _ids:
-                res = self.player_interface.GetOwnedGames(
-                    steamid=_id, include_appinfo=1
-                )["response"]
-                self.game_icons = self.game_icons | {
-                    game["appid"]: game["img_icon_url"] for game in res.get("games", [])
-                }
-        response = self.user_interface.GetPlayerSummaries(steamids=_ids)
-        players = {
-            player["steamid"]: player
-            for player in response["response"]["players"]["player"]
-            if player["steamid"] in _ids
-        }
-        for k in players:
-            data = self.player_interface.GetSteamLevel(steamid=players[k]["steamid"])
-            players[k]["level"] = data["response"].get("player_level")
-        return players
+                # Some users might have their games hidden
+                if games := _interface.GetOwnedGames(steamid=_id, include_appinfo=1)[
+                    "response"
+                ].get("games"):
+                    self.game_icons = self.game_icons | {
+                        game["appid"]: game["img_icon_url"] for game in games
+                    }
+
+        profiles = {}
+        _profile: profile
+        for _profile in profile_batch(_ids):
+            entity_id = f"sensor.{_profile.persona}_level".replace(" ", "_").lower()
+            if (entry := reg.async_get(entity_id)) and not entry.disabled:
```

Comment:
this is weird, why are we doing this here? Since it's based on a coordinator, shouldn't we know which entities are registered and which profiles they represent?